apiserver/client/instanceconfig.go

```diff
@@ -93,14 +96,32 @@ func InstanceConfig(st *state.State, machineId, nonce, dataDir string) (*instanc
 		return nil, errors.Annotate(err, "setting up machine authentication")
 	}
 
-	icfg, err := instancecfg.NewInstanceConfig(st.ControllerTag(), machineId, nonce, modelConfig.ImageStream(),
-		machine.Series(), apiInfo,
-	)
+	osType, err := series.GetOSFromSeries(machine.Series())
 	if err != nil {
-		return nil, errors.Annotate(err, "initializing instance config")
+		return nil, errors.Trace(err)
+	}
+
+	instancePaths := paths.Nix
+	if osType == os.Windows {
+		instancePaths = paths.Windows
 	}
+
+	icfg := instancecfg.NewInstanceConfig(
+		instancePaths.Conf,
+		instancePaths.Temp,
+		instancePaths.Data,
+		instancePaths.Log,
+		instancePaths.MetricsSpool,
+		st.ControllerTag(),
+		machineId,
+		nonce,
+		modelConfig.ImageStream(),
+		osType,
```

Comment:
`NewInstanceConfig` now uses `osType`, specifically to store it so that downstream consumers of `InstanceConfig` can use it instead of re-deriving `os.OSType`. This was the goal of bringing this into `InstanceConfig` as it's something pertinent to starting instances.